### ISSUE TITLE
SMACSS architecture

### ISSUE THREAD

**Issue by ivanjaros** [zurb/foundation-sites#6995]
Hi,
I am working with Drupal 8 and it utilizes the SMACSS philosophy(https://smacss.com) which basically means that the styles are split into smaller files and are loaded only when needed, compared to the old "one-file-for-everything" approach.

I am looking at the structure of sass files(f6) and I am not sure how to achieve this since it looks to me like the smaller files only hold mixins, mostly, and everything is being mixed in the main file.

What would you recommend to achieve the SMACSS structure?

**Comment by ivanjaros** [comment]
OK, so I think I figured this out. I have basically created scss file for each include defined in foundation.scss file and in it I have included component with the same name and called it so the resulting css file would contain the content of the mixin.

There are 4-5 files for which the mixins do not exist or are named differently. I think it was drilldown vs drilldown-menu, visibility vs visibility-classes and so on. Fixing this was easy.

Then I have noticed that many scss files are missing dependencies on variables. It turned out the solution for this was always including the settings scss file with each scss file with all variables uncommented.

Then there was issue with global styles, I had to copy all variables from the file into the setting file and it solved the issue.

The last issue I faced was improper order of variables in the setting file. I had to move the global styles at the beginning of the file and base typography before tooltip, which i had to move above joyride.

So right now it compiles just fine. I would welcome if the order of these would be fixed so I would not have to do this next time and also the global styles should be included in the setting file as well.

**Comment by gakimball** [comment]
Hey friend, sorry for not replying to this sooner. Foundation 5 wasn't quite built with this kind of modularity in mind, and now that v5 is at end-of-life we won't be making any big architectural changes like this to it.

*However*, in Foundation 6 we made some good advancements in terms of how components are imported and exported. The framework itself is brought in with one `@import` statement, but it doesn't print any CSS by default. To export the default CSS for a component, you use an `@include` statement.

```scss
@import 'foundation';

@include foundation-global-styles;
@include foundation-grid-classes;
// ...
```

This makes the CSS a lot more portable; it can not only be put inside another partial (to pair it with your own CSS), it could also be namespaced within another class or ID.

We also flattened the dependency tree of the components. All components rely on global, button groups rely on buttons, and that's it. Everything else is completely self-contained.

Our [Sass documentation](http://foundation.zurb.com/sites/docs/sass.html) goes into more detail about how the code is architected.

Thanks for using Foundation! :)
